core/iwasm/common/wasm_shared_memory.c

```diff
@@ -21,6 +24,8 @@ typedef struct AtomicWaitInfo {
     korp_mutex wait_list_lock;
     bh_list wait_list_head;
     bh_list *wait_list;
+    // WARNING: insert to the list allowed only in acquire_wait_info
+    // otherwise there will be data race as described in PR #2016
```

Comment:
Had better use `/* .. */` to add the comment, here it is C source code, we usually do like that (at least in several core folders) to avoid possible compilation warning.

Reply:
Fixed